libdebug/utils/libcontext.py

```
@@ -173,6 +176,24 @@ def terminal(self: LibContext, value: list[str] | str) -> None:
 
         self._terminal = value
 
+    @property
+    def debuginfod_server(self: LibContext) -> str:
+        """Property getter for debuginfod_server.
+
+        Returns:
+            _debuginfod_server (str): the current debuginfod server.
+        """
+        return self._debuginfod_server
+
+    @debuginfod_server.setter
+    def debuginfod_server(self: LibContext, value: str) -> None:
+        """Property setter for debuginfod_server, ensuring it's a valid URL."""
+        self._debuginfod_server = value
+        if type(value) is not str or (not value.startswith("http://") and not value.startswith("https://")):
```

Should we raise the exception *before* setting the new value?

---

done in https://github.com/libdebug/libdebug/pull/206/commits/ebadd25aa3d84e4dfd77255d8136da2011a614b8